src/Slist.hs

```diff
@@ -1892,3 +1901,170 @@ insert = insertBy compare
+Slist {sList = "", sSize = Size 0}
+>>> genericTake 4 $ infiniteSlist [1..]
+Slist {sList = [1,2,3,4], sSize = Size 4}
+-}
+genericTake :: Integral i => i -> Slist a -> Slist a
+genericTake i sl@Slist{..}
+    | Size i' >= sSize = sl
+    | i' <= 0 = mempty
+    | otherwise = Slist
+        { sList = L.genericTake i sList
+        , sSize = min sSize (Size i')
+        }
+    where i' = fromIntegral i
+{-# INLINE genericTake #-}
+
+{- | @O(i) | i < n@ and @O(1) | otherwise@.
+The 'genericDrop' function is an overloaded version of 'drop', which accepts
+any 'Integral' value as the number of elements to drop.
+
+>>> genericDrop 6 $ slist "Hello World"
+Slist {sList = "World", sSize = Size 5}
+>>> genericDrop 42 $ slist "oops!"
+Slist {sList = "", sSize = Size 0}
+>>> genericDrop 0 $ slist "Hello World!"
+Slist {sList = "Hello World!", sSize = Size 12}
+>>> genericDrop (-4) $ one 42
+Slist {sList = [42], sSize = Size 1}
+
+@
+>> __drop 5 $ 'infiniteSlist' [1..]__
+Slist {sList = [6..], sSize = 'Infinity'}
+@
+
+-}
+genericDrop :: Integral i => i -> Slist a -> Slist a
+genericDrop i sl@Slist{..}
+    | i <= 0 = sl
+    | Size i' >= sSize = mempty
+    | otherwise = Slist
+        { sList = L.genericDrop i sList
+        , sSize = sSize - Size i'
+        }
+    where i' = fromIntegral i
+{-# INLINE genericDrop #-}
+
+{- | @O(i) | i < n@ and @O(1) | otherwise@.
+The 'genericSplitAt' function is an overloaded version of 'splitAt', which
+accepts any 'Integral' value as the position at which to split.
+
+>>> genericSplitAt 5 $ slist "Hello World!"
+(Slist {sList = "Hello", sSize = Size 5},Slist {sList = " World!", sSize = Size 7})
+>>> genericSplitAt 0 $ slist "abc"
+(Slist {sList = "", sSize = Size 0},Slist {sList = "abc", sSize = Size 3})
+>>> genericSplitAt 4 $ slist "abc"
+(Slist {sList = "abc", sSize = Size 3},Slist {sList = "", sSize = Size 0})
+>>> genericSplitAt (-42) $ slist "??"
+(Slist {sList = "", sSize = Size 0},Slist {sList = "??", sSize = Size 2})
+
+@
+>> __genericSplitAt 2 $ 'infiniteSlist' [1..]__
+(Slist {sList = [1,2], sSize = 'Size' 2}, Slist {sList = [3..], sSize = 'Infinity'})
+@
+
+-}
+genericSplitAt :: Integral i => i -> Slist a -> (Slist a, Slist a)
+genericSplitAt i sl@Slist{..}
+    | i <= 0 = (mempty, sl)
+    | Size i' >= sSize = (sl, mempty)
+    | otherwise =
+      let (l1, l2) = L.genericSplitAt i sList
+          s2 = sSize - Size i'
+      in (Slist l1 $ Size i', Slist l2 s2)
+    where
+      i' = fromIntegral i
+{-# INLINE genericSplitAt #-}
+
+{- | @O(i) | i < n@ and @O(1) | otherwise@.
+The 'genericAt' function is an overloaded version of 'at', which
+accepts any 'Integral' value as the position. If the element on the given
+position does not exist it will return 'Nothing'.
+
+>>> let sl = slist [1..10]
+>>> genericAt 0 sl
+Just 1
+>>> genericAt (-1) sl
+Nothing
+>>> genericAt 11 sl
+Nothing
+>>> genericAt 9 sl
+Just 10
+-}
+genericAt :: Integral i => i -> Slist a -> Maybe a
+genericAt n Slist{..}
```

Comment:
Instead of the where clause you can use `ViewPatterns`, and deal with `n` variable instead of introducing `n'`. But as
```haskell
genericAt (fromIntegral -> n) Slist{..}
```

But as you're working with `Int` you can just reuse `at` function here.